src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final int MILLIS = 300;
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+        return CompletableFuture.supplyAsync(() -> getUserDetails(userId), executor);
+    }
+
+    private UserData getUserDetails(String userId) {
+        try {
+            Thread.sleep(MILLIS);
```

Comment:
Using a fixed delay defined by the `MILLIS` constant may not be flexible for different environments or use cases. It would be better to inject this value or make it configurable.

Reply:
And again - I've added Thread.sleep(MILLIS) to simulate a working process.